client/src/modules/user/UserProfile/UserProfile.jsx

```diff
@@ -56,10 +73,61 @@ class UserProfile extends PureComponent {
     }));
   };
 
+  handleNameChange = event => {
+    const {
+      target: { value }
+    } = event;
+
+    this.setState({ userName: value, isNameUpdated: true }, this.validateName);
+  };
+
+  validateName = () => {
+    const { userName } = this.state;
+    let errorMessageId;
+
+    errorMessageId = validateWith(
+      value => !isEmpty(value, { ignore_whitespace: true })
+    )('common.form.required-warning')(userName);
+
+    if (_trim(userName)) {
+      errorMessageId = validateWith(value =>
+        isLength(value, { min: 1, max: 32 })
+      )('common.form.field-min-max')(userName);
+    }
+
+    this.setState({ errorMessageId });
+  };
+
+  handleNameUpdate = () => {
+    const { errorMessageId, isNameUpdated, userName } = this.state;
+    const {
+      currentUser: { id: userId },
+      updateName
+    } = this.props;
+
+    if (isNameUpdated && !errorMessageId) {
+      updateName(userName, userId);
```

Comment:
isn't this an async action? Shouldn't we use `then... catch` or `async...await` with  `try-catch` and wait for the response and handle errors?  Now, the state is changed before we get response